sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -155,6 +155,17 @@ def __init__(
         super(BlobClient, self).__init__(parsed_url, service='blob', credential=credential, **kwargs)
         self._client = AzureBlobStorage(self.url, pipeline=self._pipeline)
 
+    def _format_url(self, hostname):
+        container_name = self.container_name
+        if isinstance(container_name, six.text_type):
```

Comment:
Just curious ^_^, is there a specific reason we want to convert container_name to bytes first? I think quote will also convert it to bytes using utf-8 even we delete this ` if isinstance`.

Reply:
hmm good question - IIRC it was needed for something... but I will remove and re-run the tests to confirm whether it's still needed :)